framework/src/bundle/BundleResource.cpp

```diff
@@ -61,6 +60,26 @@ class BundleResourcePrivate
   std::atomic<int> ref;
 };
 
+BundleResourcePrivate::BundleResourcePrivate(std::shared_ptr<const BundleArchive> archive)
+  : archive(std::move(archive))
+  , ref(1)
+{
+  if (this->archive) {
+    this->archive->IncrementNumOpenResources();
+  }
+}
+
+BundleResourcePrivate::~BundleResourcePrivate()
+{
+  if (this->archive) {
+    this->archive->DecrementNumOpenResources();
+
+    if (this->archive->GetNumOpenResources() == 0) {
+      this->archive->GetResourceContainer()->CloseContainer();
+    }
```

Comment:
move this logic into the ```DecrementNumOpenResources()``` and either rename the function or document it to reflect the behavior.